staging/src/kubevirt.io/api/instancetype/v1alpha1/types.go

```diff
@@ -71,19 +71,6 @@ type VirtualMachineClusterInstancetypeList struct {
 	Items           []VirtualMachineClusterInstancetype `json:"items"`
 }
 
-// VirtualMachineInstancetypeSpecRevision is used to capture a versioned copy of VirtualMachineInstancetypeSpec within a ControllerRevision
-//
-// **WARNING** This struct should never change ensuring it can always be unmarshaled from a ControllerRevision
-type VirtualMachineInstancetypeSpecRevision struct {
```

Comment:
why was this removed, and what has changed as a result? Also, what impact would this have on upgrades if someone was using an instancetype with a previous controller revision that used this structure?

Reply:
Lets discuss this in https://github.com/kubevirt/kubevirt/pull/8282 that introduced these changes, I've just included the series here as I wanted that PR to land first hopefully avoiding conflicts.

Reply:
https://github.com/kubevirt/kubevirt/pull/8282#discussion_r955229409

Reply:
@lyarwood your approach here with taking the revision early in the VM's lifecycle (similar to how DVs are handled) looks totally fine to me. If you want to revise this PR to only include those changes, i think we can get it in

Reply:
ACK I'll do that today then, thanks.

Reply:
@davidvossel Done.